### PR TITLE
Add the bamhash tool

### PR DESCRIPTION
**Pull request by remimarenco**

### CODE REVIEW

File: recipes/bamhash/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  version: 1.0
+
+source:
+  git_url: https://github.com/DecodeGenetics/BamHash.git
+  git_tag: v1.0
+
+requirements:
+  build:
+    - openssl
+    - gcc
```
Comment (bgruening) [inline recipes/bamhash/meta.yaml]: GCC is recommended for linux but not for osx. So please use this
```
- gcc # [linux]
- llvm # [osx]
```

Comment (bgruening) [inline recipes/bamhash/meta.yaml]: I guess openssl is also a runtime dependency.

Comment (remimarenco) [inline recipes/bamhash/meta.yaml]: Thanks awesome

File: recipes/bamhash/build.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+export DYLD_LIBRARY_PATH=$PREFIX/lib
+
```
Comment (bgruening) [inline recipes/bamhash/build.sh]: Try this one: `CFLAGS=-I$PREFIX/include  make`
Moreover, make install does not exists. You need to copy the binaries to `$PREFIX/bin`

Have a look here for the name of binaries: https://github.com/bgruening/galaxytools/blob/master/tools/bamhash/tool_dependencies.xml

File: recipes/bamhash/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package:
+  name: bamhash
+  version: 1.0
+
+source:
+  git_url: https://github.com/DecodeGenetics/BamHash.git
```
Comment (bgruening) [inline recipes/bamhash/meta.yaml]: https://github.com/DecodeGenetics/BamHash/archive/v1.0.tar.gz

File: recipes/bamhash/build.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+export CFLAGS=-I$PREFIX/include make
+
+if [ ! -d "$PREFIX/bin" ]; then
+    mkdir $PREFIX/bin
+fi
+
```
Comment (bgruening) [inline recipes/bamhash/build.sh]: export DYLD_LIBRARY_PATH=$PREFIX/lib

CFLAGS=-I$PREFIX/include  make all

mv bamhash_checksum_bam $PREFIX/bin
mv bamhash_checksum_fastq $PREFIX/bin

**Comment by remimarenco** [comment]
Weird as it ```conda build bamhash``` works on my Ubuntu... :s
@bgruening if you want to take a look...

File: recipes/bamhash/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+requirements:
+  build:
+    - gcc # [linux]
+    - llvm # [osx]
+    - openssl
+    - zlib
+
+  run:
+    - openssl
+    - libgcc
```
Comment (bgruening) [inline recipes/bamhash/meta.yaml]: This is only needed for linux, if it's needed at all.
The zlib error look wired :(
I need to look at this closer if I find time.

Comment (remimarenco) [inline recipes/bamhash/meta.yaml]: Don't be. Thanks for taking the time to resolve this together!

**Comment by remimarenco** [comment]
And build ok here too @bgruening :). I guess I need to clean the things now...it should contains quite a few useless things :)

**Comment by bgruening** [comment]
Yes please cleanup and merge :)

**Comment by bgruening** [comment]
@remimarenco ping

**Comment by remimarenco** [comment]
@bgruening pong! (thx)

**Pull request merged**